ding/framework/message_queue/perfs/perf_exchanger_p2p.py

```diff
@@ -0,0 +1,256 @@
+import time
+import torch
+import treetensor.torch as ttorch
+import argparse
+import os
+
+from queue import Empty
+from typing import TYPE_CHECKING, List, Dict
+from ditk import logging
+
+from ding.utils.comm_perf_helper import tensor_size_beauty_print, byte_beauty_print, \
+    dtype_2_byte, DO_PERF, time_perf_avg, time_perf_once, print_timer_result_csv, timeout, UNIT_SIZE_LIST
+from dataclasses import dataclass
+from ding.utils.data.structure.lifo_deque import LifoDeque
+from ding.framework.message_queue.torch_rpc import DeviceMap, RECV, CLINET_REGISTER_STUB, \
+    SERVER_REGISTER_STUB, NOTIFY_SHUTDOWN, TORCHRPCMQ, remote_callback_for_test, perf_vars, \
+    set_local_mq_test
+import torch.distributed as dist
+
+LENGTH = 5
+REPEAT = 5
+EXP_NUMS = 6
+
+
+@dataclass
+class SendInfo:
+    send_map_dict: Dict = None
+    sending_flag: int = 0
+
+
+global_send_info_dict = dict()
+
+
+# exec in remote side
+def _tensor_send(
+        peer_name: str,
+        key: str,
+        task_id: int,
+        tensor: torch.tensor,
+        nums: int,
+        send_id: int,
+        timestamp: float = None,
+        empty_params: dict = None
+) -> bool:
+    global global_send_info_dict
+    send_info_dict = global_send_info_dict
+    send_info: SendInfo
+
+    if task_id == 0:
+        send_info = SendInfo()
+        send_info.send_map_dict = dict()
+        send_info_dict[send_id] = send_info
+    else:
+        send_info = None
+        while True:
+            if send_id in send_info_dict.keys():
+                send_info = send_info_dict[send_id]
+                if send_info is not None:
+                    break
+
+    assert isinstance(send_info, SendInfo)
+
+    # In practice, it means that operations on shared variables of
+    # built-in data types (ints, lists, dicts, etc)
+    # that “look atomic” really are.
+    if key in send_info.send_map_dict.keys():
+        raise RuntimeError("Multiple state_dict's key \"{}\" received form peer \"{}\"!".format(key, peer_name))
+
+    send_info.send_map_dict[key] = tensor
+
+    if task_id == nums - 1:
+        while len(send_info.send_map_dict) != nums:
```

Comment:
add sleep here?

Reply:
Done, a very small sleep time is set,  the intention here is to implement a busy-waiting loop for very short waiting time.